### PR TITLE
refactor(self-managed): deprecate zeebe config props for identity

### LINKED ISSUE #2980: Deprecate Identity-related Zeebe configuration properties
Description

Deprecate [the Identity configuration properties](https://docs.camunda.io/docs/self-managed/zeebe-deployment/configuration/gateway-config/#zeebegatewayclustersecurityauthenticationidentity) in Zeebe.

Zeebe now integrates the `identity-spring-boot-autoconfig` dependency which can pick up Identity-specific configuration properties directly, and pass them to the Zeebe Gateway.

```[tasklist]
### AC
- [x] Add a note that Zeebe's Identity configuration properties are deprecated.
```

### Additional context
* Related to https://github.com/camunda/zeebe/issues/15389
* Epic issue: https://github.com/camunda-cloud/identity/issues/2162

### PR DESCRIPTION
**Pull request by koevskinikola**
## Description

Deprecate the Zeebe configuration properties for Camunda Identity. Zeebe now integrates the identity-spring-boot-autoconfiguration module which picks up any Identity-specific configuration properties and provides them as an IdentityConfiguration instance to Zeebe.

closes #2980

## When should this change go live?

- [x] This change is not yet live and should not be merged until https://github.com/camunda/zeebe/pull/15434 is merged
- [ ] There is no urgency with this change.
- [ ] This change or page is part of a marketing blog, conference talk, or something else on a schedule.
- [x] This functionality is already available but undocumented.
- [ ] This is a bug fix or security concern.

## PR Checklist

- [ ] I have added changes to the relevant `/versioned_docs` directory, or they are not for an **already released version**.
- [x] I have added changes to the main `/docs` directory (aka `/next/`), or they are not for **future versions**.
- [x] My changes require an [Engineering review](https://github.com/camunda/camunda-docs/blob/main/howtos/documentation-guidelines.md#review-process), and I've assigned an engineering manager or tech lead as a reviewer, or my changes do not require an Engineering review.
- [x] My changes require a [technical writer review](https://github.com/camunda/camunda-docs/blob/main/howtos/documentation-guidelines.md#review-process), and I've assigned @christinaausley as a reviewer, or my changes do not require a technical writer review.

### CODE REVIEW

**Comment by koevskinikola** [comment]
@Ben-Sheppard I've added you as reviewer for this PR in case I missed pointing to any relevant Identity docs when deprecating Zeebe's Identity config properties.

@korthout the change is only marking Zeebe's Identity config properties as deprecated. Please let me know if you don't have the capacity to review it before the `8.4.0-alpha2` release.

**Review by christinaausley — APPROVED** [review]

File: docs/self-managed/zeebe-deployment/configuration/gateway.md [hunk 1]
```diff
@@ -170,6 +170,11 @@ security:
 
 ### zeebe.gateway.cluster.security.authentication.identity
 
+:::note
+The Zeebe configuration properties for Camunda Identity are deprecated as of version `8.4.0`. Use the dedicated
+Camunda Identity properties or the [corresponding environment variables](../../identity/deployment/configuration-variables.md#core-configuration).
```
Comment (Ben-Sheppard) [inline docs/self-managed/zeebe-deployment/configuration/gateway.md:175]: Just a thought for maybe @christinaausley - here we link directly to Identity however the configuration, whilst its part of the Identity SDK is actually set within each component. Should we link here instead to the OpenID Connect docs that will be introduced with https://github.com/camunda-cloud/identity/issues/2289#issuecomment-1839081470?

**Review by korthout — CHANGES_REQUESTED** [review]
There are more references in our docs to these config properties, like [here](https://github.com/camunda/zeebe/pull/15434#discussion_r1415171853). 

❌ Please ensure all references are updated.

File: docs/self-managed/zeebe-deployment/configuration/gateway.md (hunk 1 above)
Comment (christinaausley) [reply docs/self-managed/zeebe-deployment/configuration/gateway.md:175]: Yes -- agreed here @Ben-Sheppard. Could always link to both [Camunda Identity properties] and [corresponding environment variables].

**Comment by koevskinikola** [comment]
@korthout I updated the multi-tenancy config docs, as well as a link reference in the Client Authorization section.

I wasn't able to find any other references.

**Review by korthout — DISMISSED** [review]
LGTM

**Comment by akeller** [comment]
We should announce deprecations on the [Announcements page](https://docs.camunda.io/docs/next/reference/announcements/) as well. You should see an 8.4 section on [GitHub](https://github.com/camunda/camunda-docs/pull/2982) (not yet released.)

**Comment by christinaausley** [comment]
@koevskinikola Let me know when you are ready for final review based on feedback and I'll get this approved and merged for you 👍

**Comment by koevskinikola** [comment]
@christinaausley the PR is ready (excluding the [suggestion by Ben](https://github.com/camunda/camunda-docs/pull/2981#discussion_r1415166112) since those docs are not available yet).

**Review by christinaausley — APPROVED** [review]

**Comment by akeller** [comment]
Just to clarify here, it looks like we are good to merge and release with `8.4.0-alpha2`. I will add the label. If that is not correct, please speak up.

**Pull request merged**